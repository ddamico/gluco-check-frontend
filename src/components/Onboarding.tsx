import React from "react";
import { useTranslation } from "react-i18next";
import {
  Container,
  Grid,
  Link,
  makeStyles,
  Typography,
} from "@material-ui/core";

const assistantIcon = require("../images/icon-assistant.svg");

const useStyles = makeStyles((theme) => ({
  contentGrid: {
    height: "90vh",
  },
  callResponseContainer: {
    "& .assistantBubble img": {
      width: "54px",
      height: "54px",
      display: "block",
      paddingBottom: theme.spacing(2),
      float: "right",
    },
    "& .assistantBubble-question, & .assistantBubble-response": {},
    "& .assistantBubble p": {
      clear: "both",
      marginBottom: theme.spacing(2),
      padding: "14px 30px",
      borderRadius: "62px",
      borderWidth: "2px",
      borderStyle: "solid",
    },
    "& .assistantBubble-question": {
      textAlign: "start",
      alignSelf: "flex-start",
    },
    "& .assistantBubble-question p": {
      backgroundColor: "#ccc",
      borderColor: "#ccc",
      maxWidth: "70%",
    },
    "& .assistantBubble-response": {
      textAlign: "end",
      alignSelf: "flex-end",
    },
    "& .assistantBubble-response p": {
      borderColor: "#e8eaed",
      textAlign: "inline-end",
    },
    [theme.breakpoints.down("sm")]: {
      "& .assistantBubble-response, & .assistantBubble-response": {
        fontSize: ".9rem",
      },
    },
  },
}));

const questionAndAnswer = (
  containerClass: string,
  question: string,
  answer: string,
  index: number
) => {
  return (
    <Grid container direction="column" className={containerClass} key={index}>
      <Grid item className="assistantBubble assistantBubble-question">
        <Typography variant="h5" component="p">
          Hey Google, ask Gluco Check my blood sugar
        </Typography>
      </Grid>
      <Grid item className="assistantBubble assistantBubble-response">
        <img src={assistantIcon} alt=""></img>
        <Typography variant="h5" component="p">
          6.5 and steady as of five minutes ago
        </Typography>
      </Grid>
    </Grid>
  );
};

function Onboarding() {
  const classes = useStyles();
  const { t } = useTranslation();

  const questionsAndAnswers = [
    {
      question: t("onboarding.q1"),
      answer: t("onboarding.a1"),
    },
  ];

  return (
    <Container maxWidth="md">
      {questionsAndAnswers.map((item, index) =>
        questionAndAnswer(
          classes.callResponseContainer,
          item.question,
          item.answer,
          index
        )
      )}
    </Container>
  );
}

export default Onboarding;
