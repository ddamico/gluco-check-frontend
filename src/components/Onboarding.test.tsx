import React from "react";
import { cleanup, render } from "@testing-library/react";
import { axe } from "jest-axe";
import Onboarding from "./Onboarding";

jest.mock("react-i18next", () => ({
  useTranslation: () => {
    return {
      t: jest.fn().mockImplementation((i) => {
        return i;
      }),
    };
  },
}));

afterEach(cleanup);

describe("Landing page", () => {
  it("renders the component", () => {
    const { container } = render(<Onboarding />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("has no axe violations", async () => {
    const { container } = render(<Onboarding />);
    expect(await axe(container)).toHaveNoViolations();
  });
});
