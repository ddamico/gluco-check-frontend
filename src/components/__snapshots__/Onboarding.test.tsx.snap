// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing page renders the component 1`] = `
<div
  class="MuiContainer-root MuiContainer-maxWidthMd"
>
  <div
    class="MuiGrid-root makeStyles-callResponseContainer-2 MuiGrid-container MuiGrid-direction-xs-column"
  >
    <div
      class="MuiGrid-root assistantBubble assistantBubble-question MuiGrid-item"
    >
      <p
        class="MuiTypography-root MuiTypography-h5"
      >
        onboarding.q1
      </p>
    </div>
    <div
      class="MuiGrid-root assistantBubble assistantBubble-response MuiGrid-item"
    >
      <img
        alt=""
        src="[object Object]"
      />
      <p
        class="MuiTypography-root MuiTypography-h5"
      >
        onboarding.a1
      </p>
    </div>
  </div>
</div>
`;
